core/src/main/java/nl/nn/adapterframework/validation/AbstractXmlValidator.java

```diff
@@ -207,14 +217,17 @@ public String validate(InputSource inputSource, ValidatorHandler validatorHandle
 	 * @return the result event, e.g. 'valid XML' or 'Invalid XML'
 	 * @throws XmlValidatorException, when configured to do so
 	 */
-	public String finalizeValidation(ValidationContext context, PipeLineSession session, Throwable t) throws XmlValidatorException {
+	public ValidationResult finalizeValidation(ValidationContext context, PipeLineSession session, Throwable t) throws XmlValidatorException {
 		if (t != null) {
-			return handleFailures(context.getErrorHandler(), session, XML_VALIDATOR_PARSER_ERROR_MONITOR_EVENT, t);
+			return handleFailures(context.getErrorHandler(), session, ValidationResult.PARSER_ERROR, t);
+		}
+		if (context.getErrorHandler().isErrorOccurred()) {
+			return handleFailures(context.getErrorHandler(), session, ValidationResult.NOT_VALID, null);
 		}
-		if (context.getErrorHandler().hasErrorOccured()) {
-			return handleFailures(context.getErrorHandler(), session, XML_VALIDATOR_NOT_VALID_MONITOR_EVENT, null);
+		if (context.getErrorHandler().isWarningsOccurred()) {
+			return handleFailures(context.getErrorHandler(), session, ValidationResult.VALID_WITH_WARNINGS, null);
 		}
-		return XML_VALIDATOR_VALID_MONITOR_EVENT;
+		return ValidationResult.VALID;
```

Comment:
ik zou bijna zeggen dat de handleFailures zelf de ErrorHandler op zou kunnen halen, en dan een result te bedenken.
Misschien is het het mooist om `handleFailures` te hernoemen naar `finalizeValidation` zodat het een overloaded methode wordt. Dan past het meer bij elkaar (en dan ook de methodes bij elkaar zetten)